### PR TITLE
JAMES-3405 Expose metrics over HTTP

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: server/protocols/webadmin-integration-test/webadmin-integration-test-common/src/main/java/org/apache/james/webadmin/integration/WebAdminServerIntegrationTest.java [hunk 1]
```diff
@@ -98,6 +98,19 @@ void mailQueueRoutesShouldBeExposed() {
             .statusCode(HttpStatus.OK_200);
     }
 
+    @Test
+    void metricsRoutesShouldBeExposed() {
+        String body = when()
+            .get("/metrics").prettyPeek()
```
Comment (Arsnael) [inline server/protocols/webadmin-integration-test/webadmin-integration-test-common/src/main/java/org/apache/james/webadmin/integration/WebAdminServerIntegrationTest.java:104]: debug spotted

File: server/protocols/webadmin-integration-test/webadmin-integration-test-common/src/main/java/org/apache/james/webadmin/integration/WebAdminServerIntegrationTest.java [hunk 2]
```diff
@@ -98,6 +98,19 @@ void mailQueueRoutesShouldBeExposed() {
...
+            .then()
```
Comment (Arsnael) [inline server/protocols/webadmin-integration-test/webadmin-integration-test-common/src/main/java/org/apache/james/webadmin/integration/WebAdminServerIntegrationTest.java:105]: indent

File: server/protocols/webadmin/webadmin-dropwizard-metrics/pom.xml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+            <artifactId>testing-base</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>io.dropwizard.metrics</groupId>
+            <artifactId>metrics-core</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>${james.groupId}</groupId>
+            <artifactId>metrics-tests</artifactId>
```
Comment (Arsnael) [inline server/protocols/webadmin/webadmin-dropwizard-metrics/pom.xml:58]: wrong order

**Review by Arsnael — APPROVED** [review]

File: server/protocols/webadmin/webadmin-dropwizard-metrics/pom.xml [hunk 4]
```diff
@@ -58,6 +53,11 @@
             <artifactId>metrics-tests</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>${james.groupId}</groupId>
```
Comment (Arsnael) [inline server/protocols/webadmin/webadmin-dropwizard-metrics/pom.xml:57]: nope, james module, should be before third party modules :)

**Review by Arsnael — APPROVED** [review]

**Review by LanKhuat — APPROVED** [review]

**Comment by chibenwa** [comment]
`MemorySpamAssassinContractTest.imapCopiesToSpamMailboxShouldBeConsideredAsSpam » ConditionTimeout`

test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**